homework/index.js

```diff
@@ -15,32 +38,127 @@
     xhr.onerror = () => cb(new Error('Network request failed'));
     xhr.send();
   }
-
-  function createAndAppend(name, parent, options = {}) {
-    const elem = document.createElement(name);
-    parent.appendChild(elem);
-    Object.keys(options).forEach(key => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.textContent = value;
-      } else {
-        elem.setAttribute(key, value);
+  // *********************************************************
+  function ShowContributions(urlText) {
+    const infoContainer = document.getElementById('info_container');
+    const infoRight = createAndAppend('div', infoContainer, {
+      class: 'info-right',
+      text: 'Contributions:',
+    });
+    fetchJSON(urlText, (err, contributors) => {
+      if (err) {
+        renderError(err);
+        return;
       }
+      contributors.forEach(contributor => {
+        const contributorLink = createAndAppend('a', infoRight, {
+          class: 'contributor-link',
+          href: contributor.html_url,
+          target: '_blank',
+        });
+        const imgDiv = createAndAppend('div', contributorLink, { class: 'contributor-img-div' });
+        createAndAppend('img', imgDiv, {
+          class: 'contributor-img',
+          src: contributor.avatar_url,
+          alt: 'Contributor photo',
+          width: '50px',
+        });
+        const dataDiv = createAndAppend('div', contributorLink, { class: 'contributor-data-div' });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-name',
+          text: contributor.login,
+        });
+        createAndAppend('div', dataDiv, {
+          class: 'contributor-contributions',
+          text: contributor.contributions,
+        });
+        createAndAppend('hr', infoRight, {
+          class: 'hr-contributor',
+        });
+      });
     });
-    return elem;
   }
-
-  function main(url) {
+  // *************************************************
+  function ShowRepoInfo(arrayOfObjects, indexNumber) {
```

Comment:
1. camelCase: `showRepoIInfo`.
2. You are passing too much information to this function. It only needs to know about the selected repository (you called it `wantedRepo`. In general, you should pass just enough info to functions to do their, not anything more than that (i.e. on a need-to-know basis). This helps to make your code more robust as the called function then cannot mess up the data is not supposed to have access to.

In this case, your function could be defined as:

```js
function showRepoInfo(wantedRepo) {
```

Then, when you call this function, for instance in line 158 you could call it like:

```js
showRepoInfo(repositories[selectEl.value]);
```

Reply:
Yes I agree 100% and I have been thinking about that form yesterday .